### PR TITLE
[release] use bazel to run release test

### PR DESCRIPTION
**Pull request by can-anyscale**
Make the release test infra runnable with bazel. This also means that the package can run as a standalone installation.

Test:
- CI
- release: https://buildkite.com/ray-project/release/builds/13363

### CODE REVIEW

File: .buildkite/release.rayci.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ steps:
       - skip-on-premerge
       - oss
     commands:
-      - ./release/run_release_test.sh microbenchmark.aws --report 
+      - bazel run //release:run_release_test -- microbenchmark.aws --report
```
Comment (aslonnie) [inline .buildkite/release.rayci.yml:11]: this will by pass a lot of logic in the shell script `run_release_test.sh`?

Comment (can-anyscale) [reply .buildkite/release.rayci.yml:11]: Yes, let me keep the shell script for ray for now; the shell script has the following additional logic:
- copy release outputs to artifacts
- retries

Both are non-critical; i mainly do this so i can reuse the release test framework on llvm-forge which doesn't require those two logics. Overtime, I can move those two logics into the py_binary itself to make the stack simpler.

Comment (aslonnie) [reply .buildkite/release.rayci.yml]: the script also handles cancellation on sigterm/sigint?

can we call bazel on this `py_binary` in the shell script too?

File: .buildkite/release.rayci.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ steps:
       - skip-on-premerge
       - oss
     commands:
-      - ./release/run_release_test.sh microbenchmark.aws --report 
+      - ./release/run_release_test.sh -- microbenchmark.aws --report
```
Comment (aslonnie) [inline .buildkite/release.rayci.yml:11]: will this added `--` work? this seems like a bazel run thing.

File: .buildkite/release.rayci.yml (hunk 1 above)
Comment (can-anyscale) [reply .buildkite/release.rayci.yml]: yes let me do that; i do want to get rid of this shell script layer, but I can do that in a follow up

**Comment by can-anyscale** [comment]
bundle the run_release_test.sh as well

File: release/BUILD.bazel [hunk 3]
```diff
@@ -384,6 +384,22 @@ py_test(
 # RELEASE TEST INFRA unit tests
 ####
 
+sh_binary(
+    name = "run",
+    srcs = ["run_release_test.sh"],
+    data = ["run_release_test"],
+    env = {
+        "NO_INSTALL": "1",
```
Comment (can-anyscale) [inline release/BUILD.bazel:392]: we set this to 1 because we don't want to install python dependencies when running with bazel

**Comment by can-anyscale** [comment]
tests are failing

**Review by aslonnie — APPROVED** [review]

**Pull request merged**